### PR TITLE
systemctl: refuse to acquire dbus connection with --global

### LINKED ISSUE #26402: `systemctl --global show` uses units only known in user scope
### systemd version the issue has been seen with

253

### Used distribution

Arch Linux

### Linux kernel version used

_No response_

### CPU architectures issue was seen on

None

### Component

systemctl

### Expected behaviour you didn't see

`systemctl --global show only-in-user-scope.service` shows `LoadState=not-found` along with other default properties.

### Unexpected behaviour you saw

It shows properties from units that only exist in current user's scope.

### Steps to reproduce the problem

1. Create a unit under the current user's scope (`~/.config/systemd/user/`)
2. `systemctl --global show unit`
3. Check for the shown properties (e.g. `LoadState`)

### Additional program output to the terminal or log subsystem illustrating the issue

_No response_

### LINKED ISSUE #26754: OSS-Fuzz issue 56915
OSS-Fuzz has found a bug in this project. Please see https://oss-fuzz.com/testcase?key=6743594502586368 for details and reproducers.

This issue is mirrored from https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=56915 and will auto-close if the status changes there.

If you have trouble accessing this report, please file an issue at https://github.com/google/oss-fuzz/issues/new.

### PR DESCRIPTION
**Pull request by yuwata**
Maybe, better to check the runtime scope each verb for better log message, but this is a good start point to not trigger assertion.

Fixes oss-fuzz#56915 (https://bugs.chromium.org/p/oss-fuzz/issues/detail?id=56915).
Fixes #26402.
Fixes #26754.

### CODE REVIEW

**Comment by YHNdnzj** [comment]
This also fixes
https://github.com/systemd/systemd/issues/26402 I suppose 😉

**Comment by yuwata** [comment]
> This also fixes #26402 I suppose wink

Indeed.
```
$ build/systemctl show --global
--global is not supported for the operation.
```

File: src/systemctl/systemctl-util.c [hunk 1]
```diff
@@ -36,6 +36,9 @@ int acquire_bus(BusFocus focus, sd_bus **ret) {
         assert(focus < _BUS_FOCUS_MAX);
         assert(ret);
 
+        if (!IN_SET(arg_runtime_scope, RUNTIME_SCOPE_SYSTEM, RUNTIME_SCOPE_USER))
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "--global is not supported for the operation.");
```
Comment (ldv-alt) [inline src/systemctl/systemctl-util.c]: This function calls either `bus_connect_transport()` or `bus_connect_transport_systemd()`,
so I had a look how each of them handles `RUNTIME_SCOPE_GLOBAL`.

`bus_connect_transport()`:
* `arg_transport == BUS_TRANSPORT_REMOTE`: returns -EOPNOTSUPP,
* `arg_transport != BUS_TRANSPORT_REMOTE`: raises assertion.

`bus_connect_transport_systemd()`:
* `arg_transport == BUS_TRANSPORT_LOCAL`: raises assertion,
* `arg_transport != BUS_TRANSPORT_LOCAL`: returns -EOPNOTSUPP.

From this I conclude that the change is technically correct.

I'd suggest a little change of the wording: "--global is not supported for _this_ operation."
Maybe `EOPNOTSUPP` would be a bit more appropriate error code, but I'm not sure.

**Review by ldv-alt — APPROVED** [review]

File: src/systemctl/systemctl-util.c (hunk 1 above)
Comment (yuwata) [reply src/systemctl/systemctl-util.c]: Thank you. The suggestion is applied.

**Pull request merged**